src/util/bip32.rs

```diff
@@ -180,13 +180,10 @@ impl FromStr for ChildNumber {
     type Err = Error;
 
     fn from_str(inp: &str) -> Result<ChildNumber, Error> {
-        Ok(match inp.chars().last().map_or(false, |l| l == '\'' || l == 'h') {
-            true => ChildNumber::from_hardened_idx(
-                inp[0..inp.len() - 1].parse().map_err(|_| Error::InvalidChildNumberFormat)?
-            )?,
-            false => ChildNumber::from_normal_idx(
-                inp.parse().map_err(|_| Error::InvalidChildNumberFormat)?
-            )?,
+        Ok(if inp.chars().last().map_or(false, |l| l == '\'' || l == 'h') {
+            ChildNumber::from_hardened_idx(inp[0..inp.len() - 1].parse().map_err(|_| Error::InvalidChildNumberFormat)?)?
```

Comment:
What is our line length limit?

This also feels harder to read than the original.

Reply:
I felt like both this and the original are unreadable, but I didn't want to actually format code in this PR, I think that after formatting the if will be more readable

Reply:
Default Rust line length limit is 100 chars. That's also what I used in https://github.com/rust-bitcoin/rust-bitcoin/pull/290.

Reply:
How would people feel about dropping it to 80?

Reply:
Personally I don't like short lines. especially in function declarations/call. I hated this so much in PEB8 when I wrote python.
I think we should use common sense. 
when you do stuff like the builder pattern than adding a space for every function is good, when you declare a function than passing the max length by 5-10 chars is fine in my opinion, we all have FHD/QHD+ displays here.

Anyway this is a rustfmt thing, not a clippy thing.

Reply:
> we all have FHD/QHD+ displays here

But some of us like to view multiple files at once ;)

Anyway sure, I'm fine with 100 lines. I'm just annoyed when rustfmt does stuff like collapsing a "common sense" multi-line builder pattern into a 100-line mess.

Reply:
yeah, even in my personal projects where I have a `rustfmt.toml` file I still find myself manually overriding stuff that it do

Reply:
@apoelstra It's possible to tweak this, I belive. `use_small_heuristics=off` https://rust-lang.github.io/rustfmt/ ?

Reply:
I use `use_small_heuristic = "Off"` in all my projects. I don't like rustfmt to "heuristically" decide whether a line "feels" short/long.

I'm personally not in favor of shortening it to 80, though. Sadly rustfmt doesn't support a separate line length for comments, where I do prefer 80.

80 is just really too short for any method declaration involving `Result`..